### PR TITLE
Update representative gene - Partly fix #2835 and #2826

### LINKED ISSUE #2835: Scout cancer interface problems (partly duplicated)
I'm copying the text from a customer ticket by Emma Tham:

Dear Scout developers,

I have used the new SCOUT layout for the first time today and have a few issues I would like to give you feedback on.

- [ ] Only one variant is annotated in the summary and it is not always the relevant variant. We are usually interested in the coding variants. See eg pat KI/Brehmer account 091: 37-DS-a, variant [37]1:g.45805919G→C where the non-coding TESK2 is listed, while the MUTYH variant which is the coding variant is not (and I need to scroll down to find it). 

I suggest that if it is not possible to select the coding one, that all variants are listed in the summary.

- [x] The ClinVar database (which is vital for our interpretation) should be clearly visible in the summary with the classification. See eg [37]3:g.178938934G→A in PIK3CA (same patient as above). This variant is a known pathogenic variant and this information should be clearly visible in the summary (it is now only visible on the first page).

- [x]  I would like to be able to dismiss variants from the first page, The functionality seems to be there, but is not active for some reason. This means that I have to open up each variant in a separate window/tab in order to dismiss it even when I can see from the first page that it is not relevant. 

I hope that these suggestions are possible?!

Thank you in advance,

Best regards

Emma

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

This is an alternative to #2830 that instead retains the idea of only showing one gene symbol in summary, using the same logic as in variantS list (most severe SO_TERM consequence). This should be understandable to most users: "Why is this weird research gene with a stopgain in my list? Ah, ok, it is also a synonymous variant in a gene in my panel. That's fine, moving on."

I agree the jinja is getting a bit hard to maintain there. We could consider moving the "primary transcipt" logic into the controller, but I don't do it here.

**How to test**:
1. compare #2830 - note that the same gene as in the variantS list should be selected as representative for the summary

Screenshot 2021-09-02 at 09 44 49

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN, CR

### CODE REVIEW

**Comment by dnil** [comment]
Right, the representative could be without NM_ as well. After lunch..

**Comment by northwestwitch** [comment]
Testing this on stage with a multi-gene variant, for instance this one:

https://scout-stage.scilifelab.se/cust000/balsamic-validation-gmsmyeloid-tumor-only-2/cancer/ac4fa05770c2e462ed36d23fe673ea4e?cancer=yes

On my branch (#2830) it would look like this:
image

On this branch:

image


I think it works as it's supposed to, bit I honestly don't know if this is what a user might want to see, can't we ask some real user?

The branch is installed on stage now.  

One thing I would add is tooltips explaining a bit that you are seeing the most representative gene instead of all genes. And also a tooltip on the transcript explaining why that tx.

File: scout/server/blueprints/variant/utils.py
Comment (northwestwitch) [inline scout/server/blueprints/variant/utils.py:55]: Nice with the shared code!

**Review by northwestwitch — APPROVED** [review]
It works fine, but I'd add tooltips and ask for feedback before merging!

**Comment by dnil** [comment]
I'm convinced this is what users believe they want to see, though there are definitely edge cases lurking, like say the variant is a known pathogenic regulatory variant in a gene in your panel, but also affects another protein with no disease linked. These are however not abundant, we have tooltips, and all the info is actually in the page after all.

**Comment by dnil** [comment]
Your test variant is actually touching such an edge case for cancer, where the ESR variant, while uninteresting, is the one that the user should evaluate first here, before possibly considering the secondary finding from the SYNE2 change.

**Comment by dnil** [comment]
But note that both users that commented actually want exactly this: ˜"the gene on variantS is right" and ˜"we usually pick the one with the most severe impact". 😸

**Comment by northwestwitch** [comment]
> But note that both users that commented actually want exactly this: ˜"the gene on variantS is right" and ˜"we usually pick the one with the most severe impact". 😸

Ok!

**Comment by northwestwitch** [comment]
@dnil is this ready for merging or do we want to test it more on stage?

**Comment by dnil** [comment]
It's always good to test a bit more. I'll put it on stage.

**Comment by dnil** [comment]
Ok, looks good. I'll merge when the tests finish!

**Pull request merged**
